src/main/java/core/basesyntax/utilities/FigureSupplier.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.utilities;
+
+import core.basesyntax.figure.core.Circle;
+import core.basesyntax.figure.core.Figure;
+import core.basesyntax.figure.core.IsoscelesTrapezoid;
+import core.basesyntax.figure.core.Rectangle;
+import core.basesyntax.figure.core.RightTriangle;
+import core.basesyntax.figure.core.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RANDOM_RANGE = 10;
+    private static final int NUMBER_OF_TYPES_OF_FIGURE = 5;
+
+    public Figure getRandomFigure() {
+
+        Random random = new Random();
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Figure returnFigure;
+        switch (random.nextInt(NUMBER_OF_TYPES_OF_FIGURE)) {
+            case (0):
+                returnFigure =
+                        new Circle(colorSupplier.getRandomColor(), random.nextInt(RANDOM_RANGE));
```

Comment:
let's create private methods for each figure creation and there explicitly create needed variables to pass to a constructor. then we can return the result of these methods right from the switch-case, without `returnFigure` variable